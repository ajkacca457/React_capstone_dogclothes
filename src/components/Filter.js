import React from 'react';
import PropTypes from 'prop-types';

const Filter = ({ changeFilter }) => (
  <div className="filter">
    <input
      type="text"
      className="form-control"
      placeholder="Search Shows"
      onChange={e => changeFilter(e.target.value)}
    />
  </div>
);

Filter.propTypes = {
  changeFilter: PropTypes.func.isRequired,
};

export default Filter;
