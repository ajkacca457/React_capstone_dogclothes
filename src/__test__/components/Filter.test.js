/*eslint-disable*/
import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import Filter from '../../components/Filter';

Enzyme.configure({ adapter: new Adapter() })

const setup = () => shallow(<Filter />);

describe('Filter component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });

  it('Should check visibility of component', () => {
    const element = component.find('.form-control');
    expect(element).toBeVisible;
  });

  it('Should check lenght of component', () => {
    const element = component.find('.form-control');
    expect(element.length).toEqual(1);
  });

  it('Should check existance of component', () => {
    const element = component.find('.form-control');
    expect(element).toBeTruthy;
  });

  it('Should check correct component renders', () => {
    const element = component.find('.search');
    expect(element).toBeFalsy;
  });

  test('renders the form-control in the page', () => {
    const item = component.find('.form-control');
    expect(item).toMatchSnapshot();
  });
});
