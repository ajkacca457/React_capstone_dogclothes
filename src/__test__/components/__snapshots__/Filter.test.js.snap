// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders the form-control in the page 1`] = `ShallowWrapper {}`;
